src/cim/components/Agent/AgentsSelectionTable.tsx

```diff
@@ -111,13 +122,21 @@ const AgentsSelectionTable: React.FC<AgentsSelectionTableProps> = ({
 
   const paginationProps = usePagination(hosts.length);
 
+  const actionChecks = React.useMemo(
+    () => ({
+      canEditDisks: (host: Host) =>
+        canEditDisksUtil(cluster ? cluster.status : undefined, host.status),
+    }),
+    [cluster],
+  );
+
   return (
     <HostsTable
       hosts={hosts}
       content={content}
       selectedIDs={selectedIDs}
       actionResolver={actionResolver}
-      ExpandComponent={DefaultExpandComponent}
+      ExpandComponent={DefaultExpandComponentAgent(onDiskRole, actionChecks.canEditDisks)}
```

Comment:
canEditDisks and onDiskRole should be part of actionResolver. actionResolver maps onDiskRole(hostId, diskId, roleValue) to onEditDiskRole(agent). 

ExpandComponent should be defined inside the AgentSelectonTable component like this:
```
const ExpandComponent = React.useCallback(
    ({ obj }) => {
      return (
        <HostDetail
          host={obj}
          onDiskRole={actionResolver.onDiskRole}
          canEditDisks={actionResolver.canEditDisks}
        />
      );
    },
    [AdditionalNTPSourcesDialogToggleComponent],
  );

Reply:
I just don't understand this part. I can't find the method onEditDiskRole(agent)